transcom/mymove
# INT B-19901 update moves table with locked moves columns

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19901)

## Summary

As we ramp up with the feature that will allow for moves to be locked while an office user is working on them, we need to set up the database to store 1) who has the move locked and is working on it, 2) store an expiration value to ensure that moves don't get forever locked.

This PR does the following:
- updates the `moves` table with two columns
- `locked_by` that references the `office_users` table
- `lock_expires_at` which sets the expiration that the move will then become unlocked, pending the office user does not unlock it prior to then (future logic to be added in BE stories)

### How to test

1. Pull this branch
2. Run `make server_run` to fire up migrations
3. If that doesn't do it, run the nuke command `make db_dev_fresh` to run all migrations
4. Ensure that the columns `locked_by` and `lock_expires_at` exist in the `moves` table along with comments

## Screenshots
![Screenshot 2024-05-07 at 8 54 36 AM](https://github.com/transcom/mymove/assets/136510600/9cc29315-e7bd-4909-92fd-83097a52d120)
